eras/conway/impl/src/Cardano/Ledger/Conway/Era.hs

```diff
@@ -52,20 +52,24 @@ type instance Value (ConwayEra c) = MaryValue c
 -------------------------------------------------------------------------------
 
 type instance EraRule "UPEC" (ConwayEra c) = VoidEraRule "UPEC" (ConwayEra c)
+type instance EraRuleFailure "UPEC" (ConwayEra c) = VoidEraRule "UPEC" (ConwayEra c)
+type instance EraRuleEvent "UPEC" (ConwayEra c) = VoidEraRule "UPEC" (ConwayEra c)
 
 type instance EraRule "NEWPP" (ConwayEra c) = VoidEraRule "NEWPP" (ConwayEra c)
+type instance EraRuleFailure "NEWPP" (ConwayEra c) = VoidEraRule "NEWPP" (ConwayEra c)
+type instance EraRuleEvent "NEWPP" (ConwayEra c) = VoidEraRule "NEWPP" (ConwayEra c)
 
 type instance EraRule "PPUP" (ConwayEra c) = VoidEraRule "PPUP" (ConwayEra c)
 type instance EraRuleFailure "PPUP" (ConwayEra c) = VoidEraRule "PPUP" (ConwayEra c)
+type instance EraRuleEvent "PPUP" (ConwayEra c) = VoidEraRule "PPUP" (ConwayEra c)
 
 type instance EraRule "MIR" (ConwayEra c) = VoidEraRule "MIR" (ConwayEra c)
 type instance EraRuleFailure "MIR" (ConwayEra c) = VoidEraRule "MIR" (ConwayEra c)
+type instance EraRuleEvent "MIR" (ConwayEra c) = VoidEraRule "MIR" (ConwayEra c)
```

Comment:
How come we don't have these EraRuleEvent type instances for other eras, for which we have an event, so like MirEvent in Shelley?  Or conversely, why do we need them here?

Reply:
Ideally they should be defined for all rules in all eras, but right now I only defined `EraRuleEvent` for events that needed it.

Reply:
I'll add the others as well so nobody gets confused by this.